pkg/policymutation/policymutation.go

```diff
@@ -403,8 +403,33 @@ func CanAutoGen(policy *kyverno.ClusterPolicy, log logr.Logger) (applyAutoGen bo
 			return false, "none"
 		}
 
-		if (len(match.Kinds) > 1 && utils.ContainsString(match.Kinds, "Pod")) ||
-			(len(exclude.Kinds) > 1 && utils.ContainsString(exclude.Kinds, "Pod")) {
+		for _, value := range match.Any {
+			if value.Name != "" || value.Selector != nil {
```

Comment:
Can we add the check for `annotation`? See https://kubernetes.slack.com/archives/CLGR9BJU9/p1632977788101600?thread_ts=1632936910.096800&cid=CLGR9BJU9.